tutorials/scripting/c_sharp/c_sharp_differences.rst

```diff
@@ -177,42 +177,203 @@ Example:
+strip_escapes            StringExtensions.StripEscapes
+substr                   StringExtensions.Substr (Consider using `string.Substring`_ or `string.AsSpan`_)
+to_ascii_buffer          StringExtensions.ToASCIIBuffer (Consider using `System.Text.Encoding.ASCII.GetBytes`_)
+to_camel_case            StringExtensions.ToCamelCase
+to_float                 StringExtensions.ToFloat (Consider using `float.TryParse`_ or `double.TryParse`_)
+to_int                   StringExtensions.ToInt (Consider using `int.TryParse`_ or `long.TryParse`_)
+to_lower                 `string.ToLower`_
+to_pascal_case           StringExtensions.ToPascalCase
+to_snake_case            StringExtensions.ToSnakeCase
+to_upper                 `string.ToUpper`_
+to_utf16_buffer          StringExtensions.ToUTF16Buffer (Consider using `System.Text.Encoding.UTF16.GetBytes`_)
+to_utf32_buffer          StringExtensions.ToUTF32Buffer (Consider using `System.Text.Encoding.UTF32.GetBytes`_)
+to_utf8_buffer           StringExtensions.ToUTF8Buffer (Consider using `System.Text.Encoding.UTF8.GetBytes`_)
+trim_prefix              StringExtensions.TrimPrefix
+trim_suffix              StringExtensions.TrimSuffix
+unicode_at               `string[int]`_ indexer
+uri_decode               StringExtensions.URIDecode (Consider using `System.Uri.UnescapeDataString`_)
+uri_encode               StringExtensions.URIEncode (Consider using `System.Uri.EscapeDataString`_)
+validate_node_name       StringExtensions.ValidateNodeName
+xml_escape               StringExtensions.XMLEscape
+xml_unescape             StringExtensions.XMLUnescape
+=======================  ==============================================================
+
+List of Godot's PackedByteArray methods that create a String and their C# equivalent:
+
+=========================  ==============================================================
+GDScript                   C#
+=========================  ==============================================================
+get_string_from_ascii      StringExtensions.GetStringFromASCII (Consider using `System.Text.Encoding.ASCII.GetString`_)
+get_string_from_utf16      StringExtensions.GetStringFromUTF16 (Consider using `System.Text.Encoding.UTF16.GetString`_)
+get_string_from_utf32      StringExtensions.GetStringFromUTF32 (Consider using `System.Text.Encoding.UTF32.GetString`_)
+get_string_from_utf8       StringExtensions.GetStringFromUTF8 (Consider using `System.Text.Encoding.UTF8.GetString`_)
+hex_encode                 StringExtensions.HexEncode (Consider using `System.Convert.ToHexString`_)
+=========================  ==============================================================
+
+* Strings are immutable in .NET, so all methods that manipulate a string don't
+  modify the original string and return a newly created string with the
+  modifications applied. To avoid creating multiple string allocations consider
+  using a `StringBuilder`_.
```

Comment:
I would move this paragraph to the top because the detail about immutability of strings is important.